SIL.Media/FFmpegRunner.cs

```diff
@@ -4,511 +4,512 @@
-                return FFmpegLocation;
-            FFmpegLocation = LocateFFmpeg();
-            return FFmpegLocation;
-        }
-
-        /// <summary>
-        /// ffmpeg is more of a "compile it yourself" thing, and yet
-        /// SIL doesn't necessarily want to be redistributing something
-        /// which may violate software patents (e.g. mp3) in certain countries, so
-        /// we ask users to get it themselves.
-        /// See: http://www.ffmpeg.org/legal.html
-        /// This tries to find where they put it.
-        /// </summary>
-        /// <returns>the path, if found, else null</returns>
-        static private string LocateFFmpeg()
-        {
-            //on linux, we can safely assume the package has included the needed dependency
-#if MONO
-			if (File.Exists("/usr/bin/ffmpeg"))
-				return "/usr/bin/ffmpeg";
-			else if (File.Exists("/usr/bin/avconv"))
-				return "/usr/bin/avconv";	// the new name of ffmpeg on Linux
-			else
+	///<summary>
+	/// FFmpeg is an open source media processing commandline library
+	///</summary>
+	public class FFmpegRunner
+	{
+		/// <summary>
+		/// If your app knows where FFMPEG lives, you can tell us before making any calls.
+		/// </summary>
+		public static string FFmpegLocation;
+
+		/// <summary>
+		/// Find the path to ffmpeg, and remember it (some apps (like SayMore) call ffmpeg a lot)
+		/// </summary>
+		/// <returns></returns>
+		static internal string LocateAndRememberFFmpeg()
+		{
+			if (null != FFmpegLocation) //NO! string.empty means we looked and didn't find: string.IsNullOrEmpty(s_ffmpegLocation))
+				return FFmpegLocation;
+			FFmpegLocation = LocateFFmpeg();
+			return FFmpegLocation;
+		}
+
+		/// <summary>
+		/// ffmpeg is more of a "compile it yourself" thing, and yet
+		/// SIL doesn't necessarily want to be redistributing something
+		/// which may violate software patents (e.g. mp3) in certain countries, so
+		/// we ask users to get it themselves.
+		/// See: http://www.ffmpeg.org/legal.html
+		/// This tries to find where they put it.
+		/// </summary>
+		/// <returns>the path, if found, else null</returns>
+		static private string LocateFFmpeg()
+		{
+			if (Platform.IsLinux)
+			{
+				//on linux, we can safely assume the package has included the needed dependency
+				if (File.Exists("/usr/bin/ffmpeg"))
+					return "/usr/bin/ffmpeg";
+				if (File.Exists("/usr/bin/avconv"))
+					return "/usr/bin/avconv"; // the new name of ffmpeg on Linux
+
 				return null;
-#endif
+			}
 
-#if !MONO
-            string withApplicationDirectory = GetPathToBundledFFmpeg();
+			string withApplicationDirectory = GetPathToBundledFFmpeg();
 
-            if (!string.IsNullOrEmpty(withApplicationDirectory) && File.Exists(withApplicationDirectory))
-                return withApplicationDirectory;
+			if (!string.IsNullOrEmpty(withApplicationDirectory) && File.Exists(withApplicationDirectory))
```

Comment:
Just below this line is a comment: "nb: this is sensitive to whether we are compiled against win32 or not, not just the host OS, as you might guess.
Is that (still) true? If so, what happens now that we are not doing conditional compilation?

Reply:
Ahh, thanks. I didn't see that comment.

I don't understand this comment - it doesn't seem to match what the code is doing. Depending on whether we run as 32- or 64-bit process we get a different value for ProgramFiles. But then we look in both "Program Files" and "Program Files (x86)" for ffmpeg.exe and take whatever we find. I don't think this change changed anything how the method works. Correct me if I miss something.